spec/furniture/marketplace/stripe_events_controller_request_spec.rb

```diff
@@ -3,9 +3,13 @@
 # rubocop:disable RSpec/VerifiedDoubles
 RSpec.describe Marketplace::StripeEventsController, type: :request do
   let(:marketplace) { create(:marketplace, :with_stripe_utility, stripe_account: "sa_1234", stripe_webhook_endpoint_secret: "whsec_1234") }
+  let(:marketplace_with_square) { create(:marketplace, :with_stripe_utility, stripe_account: "sa_1234", stripe_webhook_endpoint_secret: "whsec_1234", square_location_id: "123456789", square_environment: "sandbox", secrets: {square_access_token: "fake_square_access_token"}) }
   let(:space) { marketplace.space }
   let(:member) { create(:membership, space: space).member }
   let(:order) { create(:marketplace_order, :with_products, status: :pre_checkout, marketplace: marketplace) }
+  let(:person) { create(:person) }
+  let(:shopper) { create(:marketplace_shopper, person: person) }
+  let(:order_for_square) { create(:marketplace_order, :with_products, status: :pre_checkout, marketplace: marketplace_with_square, shopper: shopper) }
```

Comment:
FYI I did play with updating the existing `order` record at first but realized it would be simpler to consider a totally new object for the new case.

Reply:
In rspec, you can use [`context` blocks](https://rspec.rubystyle.guide/#use-contexts) to group test setup. Example:


```rb
context "when the marketplace has a Square integration" do
  let(:marketplace) {  create(:marketplace, :with_stripe_utility, stripe_account: "sa_1234", stripe_webhook_endpoint_secret: "whsec_1234", square_location_id: "123456789", square_environment: "sandbox", secrets: {square_access_token: "fake_square_access_token"})  }
  # ...
end
```

Reply:
Nice. Yeah, I like the power of the lazy evaluation offered by ruby here. Although, when possible, I prefer to put data in one place (ie the top of file) _together with_ more descriptive variable names. This will: 
- allow the developer to focus more reasily on the intent of the test in terms of it's description and commands (less code to read in the test assertion itself)
- make data easier to understand at a glance -- oh I have this _type_ of order here, and this _type_ of order there
Does that make sense?

Reply:
That said, I also sometimes rewrite larger tests suites like this that live in a single file with context blocks for every method. Then all data setup is contextual to their proper case/usage -- this improves the developer experience per what I listed above. So I'd be into making that adjustment!

Reply:
You may prefer to group the data in one place; but it is not what most people who read and write rspec will expect. In general, when I am learning a new language or framework; I try to blend in with what the community conventions are; since I've been bitten time and time again by thinking I know best 🤣